src/pages/TeamOverview.tsx

```diff
@@ -62,33 +53,56 @@ interface PageState {
 const TeamOverview = () => {
     const location = useLocation();
     const {teamId} = useParams();
-    const [pageData, setPageData] = React.useState<PageState>({});
+    const [pageData, setPageData] = React.useState<PageState>({teamLead: null, teamMembers: []});
     const [isLoading, setIsLoading] = React.useState<boolean>(true);
+    const [error, setError] = React.useState<string | null>(null);
+    const [searchTerm, setSearchTerm] = React.useState<string>('');
 
     React.useEffect(() => {
-        var getTeamUsers = async () => {
-            const {teamLeadId, teamMemberIds = []} = await getTeamOverview(teamId);
-            const teamLead = await getUserData(teamLeadId);
+        const getTeamUsers = async () => {
+            try {
+                const {teamLeadId, teamMemberIds = []} = await getTeamOverview(teamId);
+                const teamLead = await getUserData(teamLeadId);
+                const teamMembers = await Promise.all(teamMemberIds.map(id => getUserData(id)));
 
-            const teamMembers = [];
-            for(var teamMemberId of teamMemberIds) {
-                const data = await getUserData(teamMemberId);
-                teamMembers.push(data);
+                setPageData({
+                    teamLead,
+                    teamMembers,
+                });
+            } catch (err) {
+                setError('Failed to fetch team users, please reload the page to retry');
+            } finally {
+                setIsLoading(false);
```

Comment:
Remove loader even if there is an error, so that we can display either user data ou error message